### PR TITLE
Add REST API to retrieve subscriber-info

### PR DESCRIPTION
**Pull request by vithu30**
This PR adds new Publisher REST API to retrieve subscriber-info. REST API path is : `/subscriptions/{subscription-id}/subscriber-info`

Request format:
`curl -X GET "https://localhost:9443/api/am/publisher/v1/subscriptions/{subscriptionId}/subscriber-info" -H  "Authorization: Bearer xxxxxxx"`

Sample request:
`curl -X GET "https://localhost:9443/api/am/publisher/v1/subscriptions/04b306b6-49da-49fc-88ae-feb6dc87150b/subscriber-info" -H  "Authorization: Bearer 33f03b51-6f97-3076-9eb0-0045b04e1706" `

Sample response:
```
{
  "name": "user1@test.com",
  "claims": [
    {
      "name": "First Name",
      "value": "user1",
      "URI": "http://wso2.org/claims/givenname"
    },
    {
      "name": "Last Name",
      "value": "user1",
      "URI": "http://wso2.org/claims/lastname"
    },
    {
      "name": "Email",
      "value": "user1@wso2.com",
      "URI": "http://wso2.org/claims/emailaddress"
    },
    {
      "name": "Organization",
      "value": "wso2",
      "URI": "http://wso2.org/claims/organization"
    }
  ]
}
```

Related PR - https://github.com/wso2/product-apim/pull/7906
Related Issue - https://github.com/wso2/product-apim/issues/7873

### CODE REVIEW

**Comment by isharac** [comment]
Why /subscriptions/{subscription-id}/subscriber or subscriber-info
Do we support this from both APIs?

**Comment by vithu30** [comment]
> Why /subscriptions/{subscription-id}/subscriber or subscriber-info
> Do we support this from both APIs?

No. Updated the comment.

File: components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/APIProviderImpl.java [hunk 1]
```diff
@@ -2006,6 +2008,51 @@ public void makeAPIKeysForwardCompatible(API api) throws APIManagementException
...
+    /**
+     * Returns the claims of subscriber for the given subscriber.
+     *
+     * @param subscriber The name of the subscriber to be returned
+     * @return The looked up claims of the subscriber or null if the requested subscriber does not exist
+     * @throws APIManagementException if failed to get Subscriber
+     */
+    @Override
+    public List getSubscriberClaims(String subscriber) throws APIManagementException {
+        String tenantDomain = MultitenantUtils.getTenantDomain(subscriber);
```
[the remark this answers is not in the document]
Comment (vithu30) [reply components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/APIProviderImpl.java:2032]: Will test and update

[the remark this answers is not in the document]
Comment (vithu30) [reply components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/APIProviderImpl.java:2032]: Tested and it works

File: components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/dao/ApiMgtDAO.java [hunk 2]
```diff
@@ -590,6 +590,51 @@ public void addSubscriber(Subscriber subscriber, String groupingId) throws APIMa
...
+     * @param subscriptionId
+     * @return subscriber ID
+     * @throws APIManagementException
+     */
+    private int getSubscriberIdBySubscriptionUUID(String subscriptionId) throws APIManagementException {
+        Connection connection = null;
+        ResultSet rs = null;
+        PreparedStatement ps = null;
+        int subscirberId = 0;
+        try {
```
Comment (malinthaprasan) [inline components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/dao/ApiMgtDAO.java:621]: Can we use try with resources?

Comment (vithu30) [reply components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/dao/ApiMgtDAO.java]: Fixed in 4b0356d

**Pull request merged**